src/model/src/InertialParametersHelpers.cpp

```diff
@@ -172,14 +188,24 @@ bool BBFromExternalShape(ExternalMesh* extMesh, Box& box)
         offset_bb_wrt_geom(0) = (maxX+minX)/2.0;
         offset_bb_wrt_geom(1) = (maxY+minY)/2.0;
         offset_bb_wrt_geom(2) = (maxZ+minZ)/2.0;
-
+        
+        double scaledX = extMesh->link_H_geometry.getPosition().getVal(0)*scaling_factor;
+        double scaledY = extMesh->link_H_geometry.getPosition().getVal(1)*scaling_factor;
+        double scaledZ = extMesh->link_H_geometry.getPosition().getVal(2)*scaling_factor;
```

Comment:
As you can read in URDF specs (http://wiki.ros.org/urdf/XML/link) the `scale` attribute is only relevant to the mesh, not to the link_H_geometry transform that is specified via the `visual/origin` tag, and that is already expressed in meters as all the rest of quantities in iDynTree, so there is not need to scale the transform offset.